projects/annotation-ui-lib/src/lib/viewers/annotation-pdf-viewer/contextual-toolbar/contextual-toolbar.component.ts

```diff
@@ -107,8 +109,10 @@ export class ContextualToolbarComponent implements OnInit, OnDestroy {
   }
 
   handleDeleteBtnClick() {
-    this.annotationStoreService.deleteAnnotationById(this.annotation.id);
-    setTimeout(() => {this.hideToolBar(); }, 10);
-    this.annotationStoreService.setAnnotationFocusSubject(this.annotation);
+    this.annotationStoreService.deleteAnnotationById(this.annotation.id, this.annotation.page);
+    setTimeout(() => {
```

Comment:
why are we delaying here? or are we making sure that we exist this process if the process doesn't finish in 10

Reply:
i believe it just an visual thing where you would want the annotation to disappear first and then the toolbar. i believe it might be best to use a promise.